geotiepoints/geointerpolator.py

```diff
@@ -45,62 +40,59 @@ class GeoInterpolator(Interpolator):
     along the y axis (this affects how the extrapolator behaves). If
     *chunksize* is set, don't forget to adjust the interpolation orders
     accordingly: the interpolation is indeed done globaly (not chunkwise).
+
     """
-    def __init__(self, lon_lat_data, *args, **kwargs):
 
-        Interpolator.__init__(self, None, *args, **kwargs)
+    def __init__(self, lon_lat_data, *args, **kwargs):
+        super().__init__(None, *args, **kwargs)
         self.lon_tiepoint = None
         self.lat_tiepoint = None
         try:
             # Maybe it's a pyresample object ?
             self.set_tiepoints(lon_lat_data.lons, lon_lat_data.lats)
             xyz = lon_lat_data.get_cartesian_coords()
             self.tie_data = [xyz[:, :, 0], xyz[:, :, 1], xyz[:, :, 2]]
-
         except AttributeError:
             self.set_tiepoints(lon_lat_data[0], lon_lat_data[1])
-            lons_rad = np.radians(self.lon_tiepoint)
-            lats_rad = np.radians(self.lat_tiepoint)
-            x__ = EARTH_RADIUS * np.cos(lats_rad) * np.cos(lons_rad)
-            y__ = EARTH_RADIUS * np.cos(lats_rad) * np.sin(lons_rad)
-            z__ = EARTH_RADIUS * np.sin(lats_rad)
+            x__, y__, z__ = lonlat2xyz(self.lon_tiepoint, self.lat_tiepoint)
             self.tie_data = [x__, y__, z__]
 
-
-        self.new_data = []
-        for num in range(len(self.tie_data)):
-            self.new_data.append([])
-
+        self.new_data = [[]] * len(self.tie_data)
```

Comment:
Doesn't this replicate the same reference to the empty list? I think I remember something fishy with this kind of syntax.

Reply:
Good point! You are probably right I completely didn't think of that. The original for loop was flagged because the iterator variable `num` wasn't used and I changed it without actually thinking about what it was trying to accomplish.

Reply:
Well now you've done it. Turns out the GeoInterpolator was doing a lot of extra stuff it didn't need to do. Now I rewrote it to avoid all the extra stuff it was doing.